src/stapi_fastapi/routers/root_router.py

```diff
@@ -0,0 +1,150 @@
+        self.backend = backend
+        self.name = name
+        self.openapi_endpoint_name = openapi_endpoint_name
+        self.docs_endpoint_name = docs_endpoint_name
+
+        # A dict is used to track the product routers so we can ensure
+        # idempotentcy in case a product is added multiple times, and also to
+        # manage clobbering if multiple products with the same product_id are
+        # added.
+        self.product_routers: dict[str, ProductRouter] = {}
+
+        self.add_api_route(
+            "/",
+            self.get_root,
+            methods=["GET"],
+            name=f"{self.name}:root",
+            tags=["Root"],
+        )
+
+        self.add_api_route(
+            "/products",
+            self.get_products,
+            methods=["GET"],
+            name=f"{self.name}:list-products",
+            tags=["Product"],
+        )
+
+        self.add_api_route(
+            "/orders",
+            self.get_orders,
+            methods=["GET"],
+            name=f"{self.name}:list-orders",
+            response_class=GeoJSONResponse,
+            tags=["Order"],
+        )
+
+        self.add_api_route(
+            "/orders/{order_id}",
+            self.get_order,
+            methods=["GET"],
+            name=f"{self.name}:get-order",
+            response_class=GeoJSONResponse,
+            tags=["Orders"],
+        )
+
+    def get_root(self, request: Request) -> RootResponse:
+        return RootResponse(
+            links=[
+                Link(
+                    href=str(request.url_for(f"{self.name}:root")),
+                    rel="self",
+                    type=TYPE_JSON,
+                ),
+                Link(
+                    href=str(request.url_for(f"{self.name}:list-products")),
+                    rel="products",
+                    type=TYPE_JSON,
+                ),
+                Link(
+                    href=str(request.url_for(f"{self.name}:list-orders")),
+                    rel="orders",
+                    type=TYPE_JSON,
+                ),
+                Link(
+                    href=str(request.url_for(self.openapi_endpoint_name)),
+                    rel="service-description",
+                    type=TYPE_JSON,
+                ),
+                Link(
+                    href=str(request.url_for(self.docs_endpoint_name)),
+                    rel="service-docs",
+                    type="text/html",
+                ),
+            ]
+        )
+
+    def get_products(self, request: Request) -> ProductsCollection:
+        return ProductsCollection(
+            products=[pr.get_product(request) for pr in self.product_routers.values()],
+            links=[
+                Link(
+                    href=str(request.url_for(f"{self.name}:list-products")),
+                    rel="self",
+                    type=TYPE_JSON,
+                )
+            ],
+        )
+
+    async def get_orders(self, request: Request) -> list[Order]:
```

Comment:
This is the wrong type. We need a order collection because this should be returning a FeatureCollection of order Features.

Reply:
And this order collection needs a self link.

Reply:
Actually the spec doesn't define these to be features/feature collection. But it really does make sense for them to be. So we need to open an issue on the spec side to define them as such.

Reply:
We are updating the type for this PR, and going to tackle the Link portion of the work in a new issue recently opened: https://github.com/stapi-spec/stapi-fastapi/issues/92